c++/include/orc/Vector.hh

```diff
@@ -140,6 +140,11 @@ namespace orc {
      */
     virtual void resize(uint64_t capacity);
 
+    /**
+     * Heap memory used by the batch.
+     */
+    virtual int64_t memoryUse();
```

Comment:
Sure, using a separate method is cleaner than returning -1.
Just to confirm: getMemoryUsage() or getMemoryUse()?